devtools/generate_batch_config.py

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env python
+
+"""Generate a Google Batch Job configuration file.
+
+Since we're running this from a GHA runner that doesn't have our whole
+environment installed, just uses stdlib.
+
+Current shape is intended as a drop-in replacement for `gcloud compute
+instances update-container`.
+"""
+
+import argparse
+import itertools
+import json
+import logging
+from collections import OrderedDict
+from pathlib import Path
+
+logging.basicConfig()
+logger = logging.getLogger()
+
+
+def _flat(ls: list[list]) -> list:
+    return list(itertools.chain.from_iterable(ls))
+
+
+def to_config(
+    *,
+    container_image: str,
+    container_env: list[list[str]],
+    container_command: str,
+    container_arg: str,
+) -> dict:
+    """Munge arguments into a configuration dictionary."""
+    complete_env = sorted(_flat(container_env))
+    env_dict = OrderedDict(
+        (name, value.strip('"'))
+        for name, value in (pair.split("=", maxsplit=1) for pair in complete_env)
+    )
+
+    # NOTE (daz): the best documentation of the actual data structure I've found is at
+    # https://cloud.google.com/python/docs/reference/batch/latest/google.cloud.batch_v1.types.Job
+    config = {
```

Comment:
I briefly thought about doing some dataclass thing here instead of a dictionary, but that seemed *super* verbose so I abandoned that idea.